Documentation/workflow/UsingYourBuild.md

```diff
@@ -34,50 +34,122 @@ For another small walkthrough see [Dogfooding .NET Core SDK](https://github.com/
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+
+    <add key="local coreclr" value="C:\coreclr\bin\Product\Windows_NT.x64.Debug\.nuget\pkg" /> <!-- CHANGE THIS PATH to your local output path -->
+    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" /> <!-- link to corefx NuGet feed -->
+  </packageSources>
+</configuration>
+
+```
+Obviously, **you need to update path in the XML to be the path to output directory for your build**.
+
+On Windows you also have the alternative of modifying the Nuget.Config
+at `%HOMEPATH%\AppData\Roaming\Nuget\Nuget.Config` (`~/.nuget/NuGet/NuGet.Config` on Linux) with the new location.
+This will allow your new runtime to be used on any 'dotnet restore' run by the current user.
+Alternatively, you can skip creating this file and pass the path to your package directory using
+the -s SOURCE qualifier on the dotnet restore command below.   The important part is that somehow
+you have told the tools where to find your new package.
+
 ### Change project to be self-contained
 
-In order to update with your local changes the application needs to be self-contained, as opposed to running on the
+In order to update with your local changes, the application needs to be self-contained, as opposed to running on the
 shared framework.  In order to do that you will need to add a `RuntimeIdentifier` to your project.
 
-```
-  <PropertyGroup>
-    ...
-    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
-  </PropertyGroup>
+```xml
+<PropertyGroup>
+  ...
+  <RuntimeIdentifier>win-x64</RuntimeIdentifier>
+</PropertyGroup>
 ```
 
 For windows you will want `win-x64` but for other OS's you will need to set it to the most appropriate one based
 on what you built. You can generally figure that out by looking at the packages you found in your output. In our
 example you will see there is a package with the name `runtime.win-x64.Microsoft.NETCore.Runtime.CoreCLR.2.1.0-beta-25023-0.nupkg`
 so you will want to put whatever id is between `runtime.` and `Microsoft.NETCore.Runtime.CoreCLR`.
 
-Next you need to restore and publish. The publish step will also trigger a build but you can iterate on build by calling `dotnet build` as
+### Specify explicit Runtime Framework Version
+
+In order to make it work with your local core clr build you need to tell dotnet cli which Runtime to use:
+
+```
+runtime.win-x64.Microsoft.NETCore.Runtime.CoreCLR.2.1.0-beta-25023-0.nupkg
+                                                 ^---what-you-need--^
+```
+
+```xml
+<PropertyGroup>
+  ...
+  <RuntimeFrameworkVersion>2.1.0-preview1-26210-0</RuntimeFrameworkVersion>
+</PropertyGroup>
+```
+
+### Add explicit references to local runtime packages
+
+Before you restore the project you have to add two references to your `.csproj` file:
```

Comment:
This is not necessary for the quick "Update CoreCLR from raw binary output". Should this be in the later part of the doc that talks about the more involved NuGet packages path?

Reply:
it's part of the h2: ` ## Create sample self-contained application ` because it's mandatory to add these references to get sample self-contained app runing agaist local coreclr build packages.

Without adding these explicit references the dotnet publish was restoring the packages from the online feeds, not my disk.

I verified that for both `x86` and `x64`.